src/main/java/gregtech/loaders/recipe/ComponentRecipes.java

```diff
@@ -0,0 +1,387 @@
+package gregtech.loaders.recipe;
+
+import gregtech.api.recipes.RecipeMaps;
+import gregtech.api.unification.material.MarkerMaterials;
+import gregtech.api.unification.material.Materials;
+import gregtech.api.unification.ore.OrePrefix;
+import gregtech.common.items.MetaItems;
+import net.minecraftforge.fluids.FluidStack;
+
+import static gregtech.api.GTValues.L;
+
+public class ComponentRecipes {
+
+    private static final FluidStack[] pumpFluids = {Materials.Rubber.getFluid(L), Materials.StyreneButadieneRubber.getFluid((L * 3)/4), Materials.SiliconeRubber.getFluid(L/2) };
+
+    public static void register() {
+
+        //Field Generators start ---------------------------------------------------------------------------------------
+        RecipeMaps.ASSEMBLER_RECIPES.recipeBuilder()
+                .input(OrePrefix.dust, Materials.EnderPearl, 1)
+                .input(OrePrefix.circuit, MarkerMaterials.Tier.Basic, 4)
+                .fluidInputs(Materials.Osmium.getFluid(L * 2))
+                .outputs(MetaItems.FIELD_GENERATOR_LV.getStackForm())
+                .duration(100).EUt(30).buildAndRegister();
+
+        RecipeMaps.ASSEMBLER_RECIPES.recipeBuilder()
+                .input(OrePrefix.dust, Materials.EnderEye, 1)
+                .input(OrePrefix.circuit, MarkerMaterials.Tier.Good, 4)
+                .fluidInputs(Materials.Osmium.getFluid(L * 4))
+                .outputs(MetaItems.FIELD_GENERATOR_MV.getStackForm())
+                .duration(100).EUt(120).buildAndRegister();
+
+        RecipeMaps.ASSEMBLER_RECIPES.recipeBuilder()
+                .inputs(MetaItems.QUANTUM_EYE.getStackForm())
+                .input(OrePrefix.circuit, MarkerMaterials.Tier.Advanced, 4)
+                .fluidInputs(Materials.Osmium.getFluid(L * 8))
+                .outputs(MetaItems.FIELD_GENERATOR_HV.getStackForm())
+                .duration(100).EUt(480).buildAndRegister();
+
+        RecipeMaps.ASSEMBLER_RECIPES.recipeBuilder()
+                .input(OrePrefix.dust, Materials.NetherStar, 1)
+                .input(OrePrefix.circuit, MarkerMaterials.Tier.Extreme, 4)
+                .fluidInputs(Materials.Osmium.getFluid(L * 16))
+                .outputs(MetaItems.FIELD_GENERATOR_EV.getStackForm())
+                .duration(100).EUt(1920).buildAndRegister();
+
+        RecipeMaps.ASSEMBLER_RECIPES.recipeBuilder()
+                .inputs(MetaItems.QUANTUM_STAR.getStackForm())
+                .input(OrePrefix.circuit, MarkerMaterials.Tier.Elite, 4)
+                .fluidInputs(Materials.Osmium.getFluid(L * 32))
+                .outputs(MetaItems.FIELD_GENERATOR_IV.getStackForm())
+                .duration(100).EUt(7680).buildAndRegister();
+
+        //Field Generators End -----------------------------------------------------------------------------------------
```

Comment:
I don't think we need end comment as it is pretty obvious that if another section is starting previous is ending.